pkg/restapi/did/operation/operations.go

```diff
@@ -72,41 +79,66 @@ type httpClient interface {
 // GetRESTHandlers get all controller API handler available for this service
 func (o *Operation) GetRESTHandlers() []Handler {
 	return []Handler{
-		support.NewHTTPHandler(proxyURL, http.MethodGet, o.proxy),
+		support.NewHTTPHandler(resolveURL, http.MethodGet, o.resolve),
 	}
 }
 
-// Proxy will proxy requests based on provided configuration
-func (o *Operation) proxy(rw http.ResponseWriter, req *http.Request) {
+func (o *Operation) resolve(rw http.ResponseWriter, req *http.Request) {
 	did := mux.Vars(req)["did"]
 
-	log.Debugf("proxy received request for DID: %s", did)
+	log.Debugf("resolve received request for DID: %s", did)
 
-	destinationURL, err := o.ruleProvider.Transform(did)
+	result, err := o.ruleProvider.Resolve(did)
 	if err != nil {
-		o.writeErrorResponse(rw, http.StatusBadRequest,
-			fmt.Sprintf("failed to transform DID to destination URL: %s", err.Error()))
+		writeBadRequestResponse(rw, fmt.Sprintf("failed to resolve DID: %s", err.Error()))
+		return
+	}
+
+	if result.UseVDRI {
+		o.resolveWithVDRI(rw, did)
 		return
 	}
 
-	log.Debugf("proxy resolved DID '%s' to destination URL '%s'", did, destinationURL)
+	o.resolveWithProxy(rw, req, result.URL)
+}
 
+func (o *Operation) resolveWithVDRI(rw http.ResponseWriter, did string) {
+	doc, err := o.vdri.Resolve(did)
+	if err != nil {
+		writeBadRequestResponse(rw, fmt.Sprintf("failed to resolve DID: %s", err.Error()))
+		return
+	}
+
+	bytes, err := doc.JSONBytes()
+	if err != nil {
+		writeBadRequestResponse(rw, fmt.Sprintf("failed to convert DIDDoc to json bytes: %s", err.Error()))
+		return
+	}
+
+	rw.Header().Add(contentTypeHeader, didLDJson)
```

Comment:
Code from here to the end of the function is same as in resolveWithProxy - you should extract it into separate function for both calls.